clients/vulnerable-code/src/main/kotlin/VulnerableCodeService.kt

```diff
@@ -122,27 +122,14 @@ interface VulnerableCodeService {
      * list of package identifiers.
      */
     data class PackagesWrapper(
-        val packages: List<String>
+        val purls: List<String>
     )
 
     /**
      * Retrieve information about the vulnerabilities assigned to the given [packages][packageUrls].
-     * Return a map whose keys are the URLs of packages; so for each package the referenced vulnerabilities can be
-     * retrieved.
+     * Return a list with information about packages including the resolved and unresolved vulnerabilities for these
+     * packages.
      */
     @POST("api/packages/bulk_search")
-    suspend fun getPackageVulnerabilities(@Body packageUrls: PackagesWrapper): Map<String, Vulnerabilities>
-
-    /**
-     * Retrieve detail information about the specified [vulnerabilities][vulnerabilityIds]. The resulting map
-     * associates the IDs of vulnerabilities with their details.
-     */
-    @POST("api/vulnerabilities/bulk_search")
-    suspend fun getVulnerabilityDetails(@Body vulnerabilityIds: VulnerabilitiesWrapper): Map<String, Vulnerability>
-
-    /**
-     * Retrieve detail information for the vulnerability with the given [id].
-     */
-    @GET("api/vulnerabilities/{id}")
-    suspend fun getVulnerability(@Path("id") id: String): Vulnerability
```

Comment:
How about keeping these entry points, simply for completeness of the generic service client?

Reply:
I think the `getVulnerabilitiyDetails()` endpoint existed only temporarily and has been replaced by the reworked bulk API for packages (which contains vulnerability details in its response).

Regarding the `getVulnerability()` endpoint, I am not 100 % sure whether the data structures used here match exactly the ones from the bulk API - they probably should, but in any case, the data classes we use here are focused on our specific use cases and omit a number of properties we are not interested in; so this client is not that generic.

If we keep the endpoint, it should be tested, and this will increase maintenance effort.

Reply:
Ok, then let's drop them, and (re-)add only on need.